### DISCUSSION TITLE
Any way to run on Pocketbook 903 pro?

### DISCUSSION THREAD

**Discussion by TMueller83** [koreader/koreader#10318] [Q&A]
Hi, I downloaded the latest release (v2023.03) of KOReader for Pocketbook and followed the instructions to install it on my old Pocketbook 903 pro (Firmware version: 2.1.3). Whenever I try to start KOReader from the applications menu an hourglass appears on the screen for some seconds and then it disappears. Nothing else happens. I would appreciate any help to get KOReader running on my old Pocketbook 903. Thank you very much in advance!

**Comment by Frenzie** [comment]
I don't see anything that says 903 in https://github.com/koreader/koreader/blob/60849aed12375fdbc16f19fc9929179d2811ed97/frontend/device/pocketbook/device.lua

I described the basic steps here: https://github.com/koreader/koreader/issues/6311#issuecomment-650092566

> `crash.log` is a file that is automatically created when KOReader crashes. It can normally be found in the KOReader directory:
> 
> * `applications/koreader/` directory for Pocketbook

**Reply by TMueller83** [reply]
Hi @Frenzie thank you very much for your answer! The crash.log file says "Illegal Instruction" because the Pocketbook binaries provided for download in the release section are not built for the processor of the Pocketbook 903 (its an ARM 1136JF-S, as used by the Kindle-legacy devices). I assume one would need a whole new toolchain in order to target this device. I already tried to set up my own toolchain but without much success (I have little to no experience with cross compiling).

**Reply by Frenzie** [reply]
Ahh, you mean it's a device that's presumably well over a decade old. The generic Debubuntu ARM cross-compiler or for that matter some of the Kobo/Kindle binaries might just work for some quick testing for the most part, though clearly that'd be far from ideal.

**Reply by NiLuJe** [reply]
From what I can recall, you *should* just be able to use our kindle-legacy TC (e.g., `kindle` in https://github.com/koreader/koxtoolchain).

That probably means it'll never be actively supported outside of custom builds, though ;).
